torch/testing/_internal/common_nn.py

```diff
@@ -4048,6 +4062,7 @@ def padding3d_circular(input, pad):
         target_fn=lambda: torch.randn((2, 3, 4), requires_grad=True),
         reference_fn=lambda i, t, _: 1. / i.numel() *
         sum((a - b).abs().sum() for a, b in zip(i, t)),
+        check_complex=True,
```

Comment:
is the target in this case supposed to be complex or real?  The math makes it look like it should be complex, but the target created is real?

Reply:
Good catch! I checked though, and I think target_fn might be specific to test_jit, which doesn't care about check_complex or different dtypes anyway (from what I can tell from running `python test_jit -k *L1* -v` at least)

Reply:
target_fn is only used by test_jit, which basically just tries to see if scripted module behaves the same as the python module. I don't see it handling check_bfloat16 or check_half either.